### DISCUSSION TITLE
why is public key needed to clone???

### DISCUSSION THREAD

**Discussion by degenaro** [community/community#22961] [New to GitHub]
<p>[degenaro@abbey-road tmp]$ git clone <a href="mailto:git@github.com">git@github.com</a>:OpenLiberty/sample.daytrader8.git<br>
Cloning into ‘sample.daytrader8’…<br>
Warning: Permanently added the RSA host key for IP address ‘140.82.114.4’ to the list of known hosts.<br>
Permission denied (publickey).<br>
fatal: Could not read from remote repository.</p>
<p>Please make sure you have the correct access rights<br>
and the repository exists.</p>

**Comment by seveas** [comment, marked as answer]
<p>A public key is required for cloning over SSH, as the ssh protocol requires authentication. If you don’t want to have a github account or don’t want to use an ssh key, you can clone over https:</p>
<pre><code>git clone https://github.com/OpenLiberty/sample.daytrader8.git
</code></pre>

**Comment by degenaro** [comment]
<p>svn co also works.  thx.</p>
